### PR TITLE
Desktop: Fix undefined showButtonLabels

### LINKED ISSUE #2615: Desktop: Header Bar title disappears on switching themes
While switching themes, the title for menu bar items ( Toggle Sidebar, New Notebook etc.)
The titles come back when I restart the App.

[](https://gifyu.com/image/IzpE)

## Steps to reproduce

1. Open Joplin
2. Go to Appearance and change theme
3. Go back and see the bug

## Environment

Joplin version: Joplin 1.0.185 (dev, darwin)
Platform: Desktop
OS specifics: macOS

## Describe what you expected to happen

The header titles should be retained.

### PR DESCRIPTION
**Pull request by anjulalk**
Fixes #2615 

The state variable `showButtonLabels` is assigned in the function `componentDidUpdate()` which is called initially, and whenever resizing the window, etc.. But it isn't called called before rendering the component when pressing Back button from Options, which led to the variable being undefined. I think the reason is since there are no changes to the component, there isn't any reason to call the component. (If you resize the window, it gets called and button labels come back) Hence only `render()` is called.

A solution would be using `componentDidMount()` to determine if the state variable is undefined, and then assign the value. It's only called one (after the initial rendering) so there isn't a performance issue.

An alternative way is remembering the state, which is overkill IMO.

### CODE REVIEW

File: ElectronClient/gui/Header.jsx [hunk 1]
```diff
@@ -83,14 +83,13 @@ class HeaderComponent extends React.Component {
...
+		}
+	}
 
-			if (this.state.showButtonLabels !== showButtonLabels) {
-				this.setState({
-					showButtonLabels: !mediaQuery.matches,
-				});
-			}
+	componentDidMount() {
+		if (this.state.showButtonLabels == undefined) {
```
Comment (laurent22) [inline ElectronClient/gui/Header.jsx]: Do we need this conditional? Is it just an optimisation? Ideally we'd run your update logic on every mount to ensure the state is correct.

**Comment by laurent22** [comment]
Good catch, there was indeed a bug there, and your fix looks good. Could you also add showButtonLabels to the state in the constructor and set it to true by default please?

File: ElectronClient/gui/Header.jsx (hunk 1 above)
Comment (anjulalk) [reply ElectronClient/gui/Header.jsx]: I guess it's kinda unnecessary. It's only relevant at the startup where `componentDidUpdate()` is already called, so running the `determineButtonLabelState()` again is unnecessary. 

If we add `showButtonLabels: true` to the state this is not required at all.
Thanks for the suggestion. I'll update it right away.

**Comment by anjulalk** [comment]
Thanks for the input. I've updated the PR. Please have a look.

**Comment by laurent22** [comment]
All good, thanks @anjulalk!

**Pull request merged**